File: sdkwork-ai-sdk-go/api/assistant.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type AssistantApi struct {
    client *sdkhttp.Client
}

func NewAssistantApi(client *sdkhttp.Client) *AssistantApi {
    return &AssistantApi{client: client}
}

// Get assistant
func (a *AssistantApi) GetAssistant(assistant_id string) (sdktypes.AssistantResponse, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/assistants/%s", assistant_id)), nil, nil)
    if err != nil {
        var zero sdktypes.AssistantResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantResponse](raw)
}

// Update assistant
func (a *AssistantApi) UpdateAssistant(assistant_id string, body sdktypes.AssistantCreateRequest) (sdktypes.AssistantResponse, error) {
    raw, err := a.client.Post(AiApiPath(fmt.Sprintf("/assistants/%s", assistant_id)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.AssistantResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantResponse](raw)
}

// Delete assistant
func (a *AssistantApi) DeleteAssistant(assistant_id string) (sdktypes.AssistantDeleteResponse, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/assistants/%s", assistant_id)), nil, nil)
    if err != nil {
        var zero sdktypes.AssistantDeleteResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantDeleteResponse](raw)
}

// List assistants
func (a *AssistantApi) ListAssistants(query map[string]interface{}) (sdktypes.AssistantListResponse, error) {
    raw, err := a.client.Get(AiApiPath("/assistants"), query, nil)
    if err != nil {
        var zero sdktypes.AssistantListResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantListResponse](raw)
}

// Create assistant
func (a *AssistantApi) CreateAssistant(body sdktypes.AssistantCreateRequest) (sdktypes.AssistantResponse, error) {
    raw, err := a.client.Post(AiApiPath("/assistants"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.AssistantResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantResponse](raw)
}

File: sdkwork-ai-sdk-flutter/lib/src/api/assistant.dart
import '../http/client.dart';
import '../models.dart';

class AssistantApi {
  final HttpClient _client;
  
  AssistantApi(this._client);

  /// Get assistant
  Future<AssistantResponse?> getAssistant(String assistant_id) async {
    final response = await _client.get(ApiPaths.aiPath('/assistants/${assistant_id}'));
    return response is AssistantResponse ? response : null;
  }

  /// Update assistant
  Future<AssistantResponse?> updateAssistant(String assistant_id, AssistantCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/assistants/${assistant_id}'), body: body, contentType: 'application/json');
    return response is AssistantResponse ? response : null;
  }

  /// Delete assistant
  Future<AssistantDeleteResponse?> deleteAssistant(String assistant_id) async {
    final response = await _client.delete(ApiPaths.aiPath('/assistants/${assistant_id}'));
    return response is AssistantDeleteResponse ? response : null;
  }

  /// List assistants
  Future<AssistantListResponse?> listAssistants(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/assistants'), params: params);
    return response is AssistantListResponse ? response : null;
  }

  /// Create assistant
  Future<AssistantResponse?> createAssistant(AssistantCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/assistants'), body: body, contentType: 'application/json');
    return response is AssistantResponse ? response : null;
  }
}
